### PR TITLE
test: Amazon Bedrock - skip integration tests from forks

### LINKED ISSUE #689: CI: Amazon Bedrock tests fail in PRs from forks
Tests fail because of this: https://github.com/deepset-ai/haystack-core-integrations/blob/9ed323080bcc82ff70e9a0f56a6c2561fec42c2e/.github/workflows/amazon_bedrock.yml#L65-L66

Even if unit tests do not need authentication, they can't run because this step fails.

### PR DESCRIPTION
**Pull request by anakin87**
### Related Issues

- fixes #689

### Proposed Changes:

- when running from a fork, skip AWS authentication and integration tests
- slightly unrelated: also skip Mistral test if `HF_API_TOKEN` env var is not set (we can remove this once #732 is fixed)

### How did you test it?
CI: this PR comes from a fork.

### Checklist

- I have read the [contributors guidelines](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CONTRIBUTING.md) and the [code of conduct](https://github.com/deepset-ai/haystack-core-integrations/blob/main/CODE_OF_CONDUCT.md)
- I have updated the related issue with new insights and changes
- I added unit tests and updated the docstrings
- I've used one of the [conventional commit types](https://www.conventionalcommits.org/en/v1.0.0/) for my PR title: `fix:`, `feat:`, `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`.

### CODE REVIEW

File: .github/workflows/amazon_bedrock.yml [hunk 1]
```diff
@@ -62,7 +62,23 @@ jobs:
...
+      - name: Set IS_FORK environment variable (Windows)
+        if: runner.os == 'Windows'
+        run: |
+          if ($Env:GITHUB_EVENT_PULL_REQUEST_HEAD_REPO_FULL_NAME -ne $Env:GITHUB_REPOSITORY) {
+            echo "IS_FORK=true" >> $Env:GITHUB_ENV
+          }
+
+
       - name: AWS authentication
+        if: env.IS_FORK != 'true'
```
Comment (masci) [inline .github/workflows/amazon_bedrock.yml:81]: Maybe it's more complex than this, but why don't we split the tests in unit and integration, then we put `if: github.event.pull_request.head.repo.full_name == github.repository` here and in the integration tests step?

Reason is I'm not a fan of making the testing code CI dependent.

Comment (anakin87) [reply .github/workflows/amazon_bedrock.yml:81]: Makes sense. I will try!

**Comment by anakin87** [comment]
@masci I explored your idea and it works well.
Please review again when you can.

**Review by masci — APPROVED** [review]
✨

**Pull request merged**